go/internal/services/caches/caches.go

```diff
@@ -90,18 +105,70 @@ func New(config Config) (Caches, error) {
 	}
 
 	apiById, err := cache.New(cache.Config[string, db.Api]{
-		Fresh:   10 * time.Second,
-		Stale:   24 * time.Hour,
-		Logger:  config.Logger,
-		MaxSize: 1_000_000,
-
+		Fresh:    10 * time.Second,
+		Stale:    24 * time.Hour,
+		Logger:   config.Logger,
+		MaxSize:  1_000_000,
 		Resource: "api_by_id",
 		Clock:    config.Clock,
 	})
 	if err != nil {
 		return Caches{}, err
 	}
 
+	if config.CacheInvalidationTopic != nil {
+		// Create cluster caches that automatically handle distributed invalidation
+		ratelimitNamespace, err = clustering.New(clustering.Config[db.FindRatelimitNamespace]{
+			LocalCache: ratelimitNamespace,
+			Topic:      config.CacheInvalidationTopic,
+			Logger:     config.Logger,
+			NodeID:     config.NodeID,
+		})
+		if err != nil {
+			return Caches{}, err
+		}
+
+		verificationKeyByHash, err = clustering.New(clustering.Config[db.FindKeyForVerificationRow]{
+			LocalCache: verificationKeyByHash,
+			Topic:      config.CacheInvalidationTopic,
+			Logger:     config.Logger,
+			NodeID:     config.NodeID,
+		})
+		if err != nil {
+			return Caches{}, err
+		}
+
+		apiById, err = clustering.New(clustering.Config[db.Api]{
+			LocalCache: apiById,
+			Topic:      config.CacheInvalidationTopic,
+			Logger:     config.Logger,
+			NodeID:     config.NodeID,
+		})
+		if err != nil {
+			return Caches{}, err
+		}
+
+		// Start consuming invalidation events
+		consumer := config.CacheInvalidationTopic.NewConsumer()
+		consumer.Consume(context.Background(), func(ctx context.Context, event *cachev1.CacheInvalidationEvent) error {
```

Comment:
I'm not sure if this makes sense, but what if instead of having a single consumer function like here the clustering.New takes in the Consume callback and then just checks if the SourceInstance != config.NodeID and if the event.CacheName == LocalCache.Name(). 

That way we wouldn't have to keep extending the switch statement, or something along those lines atleast

Basically a OnInvalidation event for the clustering.New function?